activerecord/test/cases/batches_test.rb

```diff
@@ -164,6 +164,38 @@ def test_find_in_batches_should_ignore_the_order_default_scope
     assert_equal posts(:welcome).id, posts.first.id
   end
 
+  def test_find_in_batches_should_error_on_ignore_the_order
+    assert_raise(ArgumentError) do
+      PostWithDefaultScope.find_in_batches(error_on_ignore: true)
+    end
+  end
+
+  def test_find_in_batches_should_not_error_if_config_overriden
+    # Set the config option which will be overriden
+    ActiveRecord::Base.error_on_ignored_order_or_limit = true
```

Comment:
We should get the previous value, store in a local variable and reset it in an `ensure` block.

```
  prev = ActiveRecord::Base.error_on_ignored_order_or_limit
  ActiveRecord::Base.error_on_ignored_order_or_limit = true
  #...
ensure
  ActiveRecord::Base.error_on_ignored_order_or_limit = prev
```

This way, tests are not order dependent,